packages/components/src/navigator/navigator-screen/component.tsx

```diff
@@ -122,20 +181,40 @@ function UnconnectedNavigatorScreen(
 		locationRef.current.hasRestoredFocus = true;
 		elementToFocus.focus();
 	}, [
-		isInitialLocation,
+		skipAnimationAndFocusRestoration,
 		isMatch,
-		location.isBack,
-		location.focusTargetSelector,
-		location.skipFocus,
+		isBack,
+		focusTargetSelector,
+		skipFocus,
 	] );
 
 	const mergedWrapperRef = useMergeRefs( [ forwardedRef, wrapperRef ] );
 
-	return isMatch ? (
-		<View ref={ mergedWrapperRef } className={ classes } { ...otherProps }>
+	// Remove the screen contents from the DOM only when it not selected
+	// and its exit animation has ended.
+	if (
+		! isMatch &&
+		( exitAnimationStatus === 'idle' || exitAnimationStatus === 'animated' )
+	) {
+		return null;
+	}
+
+	return (
+		<View
+			ref={ mergedWrapperRef }
+			className={ classes }
+			onAnimationEnd={ ( e: AnimationEvent ) => {
+				if ( ! isMatch && isExitAnimation( e ) ) {
+					// When the exit animation ends on an unselected screen, set the
+					// status to 'animated' to remove the screen contents from the DOM.
+					setExitAnimationStatus( 'animated' );
+				}
+			} }
```

Comment:
I may consider adding a timeout to make sure that, if for whatever reason the correct `animationend` event doensn't fire, the timeout is going to still cause the screen to be correctly from the DOM.

Reply:
Added in https://github.com/WordPress/gutenberg/pull/64777/commits/0ae892a27f7ea9dee23efd8fd968847dec0fe078